src/main/java/com/networknt/schema/I18nSupport.java

```diff
@@ -8,10 +9,36 @@
  * @author leaves chen leaves615@gmail.com
  */
 public class I18nSupport {
+
     private static final String BASE_NAME = "jsv-messages";
-    private static ResourceBundle bundle = ResourceBundle.getBundle(BASE_NAME);
+    private static final ResourceBundle bundle;
+
+    static {
+        ResourceBundle tmpBundle = null;
+        try {
+            tmpBundle = ResourceBundle.getBundle(BASE_NAME);
+        } catch (MissingResourceException mre) {
+            // Need to avoid by all means that we fail loading ValidatorTypeCode with a
+            // "java.lang.NoClassDefFoundError: Could not initialize class com.networknt.schema.ValidatorTypeCode"
+            // due to the fact that a ResourceBundle is incomplete
+            mre.printStackTrace();
+            System.exit(1);
```

Comment:
@AndreasALoew I see here we exit if the resource bundle is not loaded correctly. Depending on how users are using this library within their application context, this might not be the optimal solution. If you have a standalone application that is doing schema validation, exit is OK. However, if this schema validator is part of the API server, we need to consider how the validator is initialized. There are two situations depending on the implementation: 

1. the schema validator is loaded during server startup. If the resource bundle is not loaded, the server stop and the user needs to fix the issue and restart the server. 
2. the schema validator is loaded on-demand when the server is handling the incoming request. In this case, stopping the server is not the right approach. We should throw the exception to make that particular request fail instead of the entire server. 

Given we might have users using the second option, we should throw the exception after print the stacktrace. What do you think?

Reply:
I agree with you @stevehu. More or less, we use the second usecase you have described and would then run into potential misbehaviour at runtime.

Reply:
I think that this discussion is somehow theoretical:

As long as before releasing any version of the validator, it is mandatory that the test suite (test cases) have been run, you will 100% notice a potential ResourceBundle inconsistency during these tests, as any inconsistency will mean that class ValidatorTypeCode cannot be loaded.

So I think we can safely assume that no version of the validator will ever be officially released that in fact would indeed come with the danger of indeed calling System.exit().

But in case you're still not happy with doing it this way, another viable alternative would be to catch and log/printStackTrace the MissingResourceException to stderr and/the configured logging solution, and then return a literal String such as "FATAL: MessageResource bundle is incomplete" from I18nSupport.getString(key) in case either bundle is null or bundle.getString(key) throws MissingResourceException.

Both these approaches still mean that any end user basically cannot properly use the validator, which is why I think that the possibility of running into this scenario in the real world mus be excluded BEFORE releasing every new version of json-schema-validator...

@stevehu @rustermi Please simply advise which alternative you'd like to see for I18nSupport - I'm perfectly fine with both...

Reply:
@AndreasALoew Thanks a lot for the explanation. If the test case can cover the scenario, then we don't need to do anything.